assisted-events-scrape/tests/integration/test_elasticsearch_storage.py

```diff
@@ -0,0 +1,175 @@
+            "timestamp": "2021-03-01T00:00:00.000Z"
+        })
+        self._es_store.store_changes(
+            index=self._index,
+            documents=docs,
+            id_fn=get_doc_cks,
+            filter_by=filter_by
+        )
+
+        # make sure the value is visible in the index
+        self._es_client.indices.refresh(index=self._index)
+
+        all_docs = self._es_client.search(index=self._index)
+
+        assert len(all_docs["hits"]["hits"]) == 1
+
+    def test_scan(self):
+        doc = {"foo": "bar"}
+        self._es_client.index(self._index, doc, refresh=True)
+
+        docs = self._es_store.scan(self._index)
+
+        assert len(list(docs)) == 1
+
+        self._es_client.index(self._index, {"barfoo": "bar"}, refresh=True)
+        self._es_client.index(self._index, {"qux": "bar"}, refresh=True)
+
+        # test that paging works
+        docs = self._es_store.scan(self._index, size=1)
+
+        assert len(list(docs)) == 3
+
+    def test_scan_with_filter(self):
+        # default mapping will create .keyword property for filtering
+        filter_by = {"foo.keyword": "bar"}
+        doc = {"foo": "bar"}
+        self._es_client.index(self._index, doc, refresh=True)
+
+        docs = self._es_store.scan(index=self._index, filter_by=filter_by)
+
+        assert len(list(docs)) == 1
+
+        self._es_client.index(self._index, {"barfoo": "bar"}, refresh=True)
+        self._es_client.index(self._index, {"qux": "bar"}, refresh=True)
+
+        docs = self._es_store.scan(self._index, size=1, filter_by=filter_by)
+
+        # it should filter out previously inserted records
+        assert len(list(docs)) == 1
+
+        self._es_client.index(self._index, {"foo": "bar", "foobar": "qux"}, refresh=True)
+        self._es_client.index(self._index, {"qux": "barfoo", "foo": "bar"}, refresh=True)
+
+        # page and filter
+        docs = self._es_store.scan(self._index, size=1, filter_by=filter_by)
+
+        assert len(list(docs)) == 3
+
+    def test_scan_source(self):
+        filter_by = {"foo.keyword": "bar"}
+        doc = {"foo": "bar"}
+        self._es_client.index(self._index, doc, refresh=True)
+
+        docs = self._es_store.scan(index=self._index, filter_by=filter_by)
+
+        assert len(list(docs)) == 1
+        # default source should not be populated
+        for doc in docs:
+            assert "foo" not in doc["_source"]
+
+        # retrieve foo field
+        docs = self._es_store.scan(index=self._index, filter_by=filter_by, source=["foo"])
+
+        # default source should not be populated
+        for doc in docs:
+            assert "foo" in doc["_source"]
+
+    def test_scan_undefined_index(self):
+        docs = self._es_store.scan(index="undefined")
+        assert 0 == len(list(docs))
+
+
+def get_doc_id(doc: dict) -> str:
+    return hashlib.sha256(json.dumps(doc).encode('utf-8')).hexdigest()
+
+
+def get_doc_cks(doc: dict) -> str:
```

Comment:
what is cks?
I prefer using the whole names instead of abbreviations, although not sure here if it's an abbreviation or an acronym

Reply:
checksum, changing it